learning-bitmark/contributing-to-bitmark/bitmark-software-development-process.md

```diff
@@ -0,0 +1,127 @@
+# Current development workflow
+
+**Right now we have these projects:**
+* bitmarkd
+* Core API
+* Updaterd
+* Backend services
+* Bitmark SDK
+
+**Development process:**
+
+Our current workflow uses OKR, where we define sprints as a deadline to deliver the software artefact, but we do not limit ourselves. We do translate the OKR in ISSUES on Github where we can track them. The external contributors can get involved with the project opening new ISSUES, being part of the discussion in existent ISSUES and make PULL REQUESTS within our projects.
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/project_bitmarkd.png)
+
+
+**Branch, Tag and release process:**
+
+We have a single development branch master, from where we create tags to perform minor and major releases. The development never happens directly on branch master, instead the developer create a development branch from master, after the developer believes its branch is ready, it opens a pull request and assign another developer to make a review. After the review process, the developer merges its branch to master branch and deletes the development branch. The developers never commit directly into the master branch without passing through the review process.
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/github_branches.png)
+
+
+**Review process:**
+
+We have a peer review process, where the developer can assign one or more individuals to verify its code, these individuals shall suggest modifications or accept the pull request, also the reviewer can reassign the review. Currently there are no time outs specified for reviewers, and in such a case, if a reviewer is delaying the review, it must discuss during the stand-up daily meeting.
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/review-process.png)
+
+# Development work-flow improvements
+
+**Development process:**
+
+This is our working flow using GitHub. Here I describe 3 basic areas of our development process.
+
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/new-review-process.png)
+
+
+**What to do when there is a new patch?**
+
+First, we need to ask ourselves, is it a trivial patch? In case it is not a 3-lines change patch, it means, it is not a trivial patch, sometimes even a 3-lines change is not trivial. We do need to be mature enough to understand what is and what is not a trivial patch. In case it is not a trivial patch, we shall open a GitHub issue with a proper title and describe what is the problem we are trying to fix, following with a pull request with a patch (if we have one), this patch must point out to the issue number. All the discussion related to implementation, shall be in the issue, code problems shall be in the pull request.
```

Comment:
I'm not entirely sure what "In case it is not a 3-lines change patch, it means, it is not a trivial patch, sometimes even a 3-lines change is not trivial." My best guess is: "Even if a patch is just a few lines long, it might not be trivial" (or something like that).

Reply:
Also, I find usage like "we need to ask ourselves" to be pretty informal. For docs addressed to the outside world, I think you want to say something like "First, Bitmark determines if it a trivial patch."